cmd/openshift-install/create.go

```diff
@@ -469,6 +474,15 @@ func waitForBootstrapConfigMap(ctx context.Context, client *kubernetes.Clientset
 	waitCtx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
 
+	// baremetal: monitor control plane bootstrapping progress
+	if platformName == baremetal.Name {
+		if err := baremetalutils.WaitForBaremetalBootstrapControlPlane(waitCtx, config); err != nil {
```

Comment:
It's not totally clear to me the reason to have the `WaitForBaremetalBootstrapControlPlane` within the `waitForBootstrapConfigMap`, as it monitors a specific bootstrap baremetal step (all the masters provisioned) and not the bootstrap cm. 
A better location could be directly in `waitForBootstrapComplete`, just before the call to `waitForBootstrapConfigMap`

Reply:
Previous [discussion](https://github.com/openshift/installer/pull/8181#discussion_r1548871383)

Reply:
If it's just a matter to share the `waitCtx` it could be created upfront and passed to both the funcs respectively. Keeping them mixed IMHO it's a source of confusion.

Reply:
OK, in the latest commit:

* I moved the `WaitForBaremetalBootstrapControlPlane` call from `waitForConfigMap` to just above to `waitForBootstrapComplete`
* I extracted the platform check and the timeout context to just before the two calls (control plane and configmap)

It's in a new commit so you can see the changes.  If you're happy with it, I can squash them.

@zaneb, any objections?

Reply:
Fine with this.

Reply:
Squashed